### Домашнее задание №1

In [1]:
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))


***1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.***

*a) Найдем сколько сочетаний по 4 карты можно выбрать из колоды, состоящей из 52 карт?*


*Воспользуемся формулой для вычисления количества сочетаний:* $C_{52}^4 = \frac{52!}{4!(52 - 4)!}$

In [2]:
combinations(52, 4)

270725

*Найдём сколько сочетаний по 4 карты крестовой масти можно выбрать из колоды, состоящей из 52 карт?*

*Количество карт масти крести в колоде равно* $\frac{52}{4} = 13$

*Воспользуемся формулой для вычисления количества сочетаний:*  $C_{13}^4 = \frac{13!}{4!(13 - 4)!}$

In [3]:
combinations(13, 4)

715

*Вероятность того, что все карты – крести:*

In [4]:
P=715/270725
P

0.0026410564225690276

*б) Рассмотрим ситуацию, когда среди 4 выбранных карт нет ни одного туза.*
* Четыре карты выбираются из 48 карт (без тузов), число таких вариантов:*

In [5]:
combinations(48, 4)

194580

Вероятность того, что среди 4-х карт не окажется ни одного туза:

In [6]:
P=194580/270725
P

0.7187367254594146

Потому как это событие противоположное, то вероятность исходного события равна:

In [7]:
P = 1 - P
P

0.2812632745405854

*Ответ: а) 0.0026410564225690276; б) 0.2812632745405855*

***2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?***

*Найдём сколько сочетаний по три кнопки существует?*

*Воспользуемся формулой для вычисления количества сочетаний:* $C_{10}^3 = \frac{10!}{3!(10 - 3)!}$

In [8]:
combinations(10, 3)

120

*Вероятность того, что человек, не знающий код, откроет дверь с первой попытки:*

In [9]:
P=1/120
P

0.008333333333333333

***Ответ: 0.008333333333333333***

***3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?***

*Найдём число сочетаний благоприятных исходов когда извлечены 3 окрашенных деталей из 9:*

*Воспользуемся формулой для вычисления количества сочетаний:* $C_{9}^3 = \frac{9!}{3!(9 - 3)!}$

In [10]:
combinations(9, 3)

84

 Число всех возможных исходов извлечения 3 деталей из 15:

In [11]:
combinations(15, 3)

455

 Искомая вероятность:

In [12]:
P=84/455
P

0.18461538461538463

***Ответ: 0.18461538461538463***

***4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?***

*Найдём число сочетаний по 2 билета из 100.*

*Воспользуемся формулой для вычисления количества сочетаний* $C_{100}^2 = \frac{100!}{2!(100 - 2)!}$

In [13]:
combinations(100, 2)

4950

Вероятность того, вероятность того, что 2 приобретенных билета окажутся выигрышными:

In [14]:
P=1/4950
P

0.00020202020202020202

***Ответ: 0.00020202020202020202***